src/frontend/apps/desk/src/features/teams/components/Member/ModalRole.tsx

```diff
@@ -0,0 +1,119 @@
+import {
+  Button,
+  Modal,
+  ModalSize,
+  Radio,
+  RadioGroup,
+  VariantType,
+  useToastProvider,
+} from '@openfun/cunningham-react';
+import { t } from 'i18next';
+import { useState } from 'react';
+
+import { Box, Text, TextErrors } from '@/components';
+import { useAuthStore } from '@/features/auth';
+import { Access, Role, useUpdateTeamAccess } from '@/features/teams/api/';
+
+interface ModalRoleProps {
+  access: Access;
+  onClose: () => void;
+  teamId: string;
+}
+
+export const ModalRole = ({ onClose, teamId, access }: ModalRoleProps) => {
+  const [localRole, setLocalRole] = useState(access.role);
+  const { userData } = useAuthStore();
+  const { toast } = useToastProvider();
+  const {
+    mutate: updateTeamAccess,
+    error: errorUpdate,
+    isError: isErrorUpdate,
+  } = useUpdateTeamAccess({
+    onSuccess: () => {
+      toast(t('The role has been updated'), VariantType.SUCCESS, {
+        duration: 4000,
+      });
+      onClose();
+    },
+  });
+
+  const isOtherOwner =
+    access.role === Role.OWNER && userData?.id !== access.user.id;
+  const isNotAllowed = isOtherOwner;
+
+  return (
+    <Modal
+      isOpen
+      leftActions={
+        <Button color="secondary" fullWidth onClick={() => onClose()}>
+          {t('Cancel')}
+        </Button>
+      }
+      onClose={() => onClose()}
+      rightActions={
+        <Button
+          color="primary"
+          fullWidth
+          onClick={() => {
+            updateTeamAccess({
+              role: localRole,
+              teamId,
+              accessId: access.id,
+            });
+          }}
+          disabled={isNotAllowed}
+        >
+          {t('Validate')}
+        </Button>
+      }
+      size={ModalSize.MEDIUM}
+      title={t('Update the role')}
+    >
+      <Box aria-label={t('Radio buttons to update the roles')}>
+        {isErrorUpdate && (
+          <TextErrors className="mb-s" causes={errorUpdate.cause} />
+        )}
```

Comment:
nit: I don't know how you will update it, but I shoot my comment:
- it makes the modal wiggle when clicking validate again
- Error message are not translated

https://github.com/numerique-gouv/people/assets/45729124/34dfa79a-d0fe-46b3-b99e-5ab38eda83b7

Reply:
Normally all the errors cases are managed in front, so the backend shouldn't emit this kind of errors.

After in real life I don't thing people will keep clicking on Validate when a error appear and notice the modal is moving.